### DISCUSSION TITLE
Task Navigation

### DISCUSSION THREAD

**Discussion by ux-fran** [nl-design-system/backlog#349] [Component suggestions]
<img width="480" height="240" alt="visuele weergave van een task navigation component" src="https://github.com/user-attachments/assets/c6b6424a-3e5b-48f9-a788-c9da5441e246" />

We willen graag meer leren over de Task Navigation component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, GitHub of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1729187095559249)

---

## Naam

Task Navigation

## Beschrijving

Overzicht van één of meerdere taken die aandacht nodig hebben, met een link om de taak uit te voeren.

## Ook bekend als

Action, Task

## Gerelateerde componenten

[Topic Navigation](https://github.com/orgs/nl-design-system/discussions/419)

## Onderzoek

[Generieke Services schetsboek - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=11-6995&p=f&t=6N0ZfdQewPcatf8v-11)
In dit Figma schetsboek is inspiratie en schetsen te vinden die het ontwerpproces van de Task Navigation component weergeven. Het bevat verschillende use cases, zoals ‘niet-verplichte taken’ en ‘verstreken deadline’.

[Generieke Services onderzoek - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=3319-237512&t=6N0ZfdQewPcatf8v-11)
In Figma is informatie verzameld vanuit verschillende organisaties, die inzichten bieden in manieren van notificeren. Dit omvat onder andere hoe een ‘taak’ (oftewel de Task Navigation component) technisch geïmplementeerd kan worden en in welke scenario’s deze effectief ingezet kan worden.

[Generieke Services concept - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=714-33964&t=6N0ZfdQewPcatf8v-11)
In Figma is uitgebreide informatie beschikbaar over de taak: hoe deze eruitziet, hoe deze werkt, en verschillende scenario’s waarin de Task Navigation component wordt gebruikt. Denk bijvoorbeeld aan de inhoud van de Task Navigation, en waar deze zich bevindt in de 'Mijn omgeving' wanneer een gebruiker een betaling aan de gemeente moet uitvoeren.
 
[Rapport Generieke Services 2023 - Gemeente Den Haag (PDF, 9 MB)](https://github.com/user-attachments/files/16993296/Rapport.generieke.services.23032023.def.pdf)
Uit dit gebruikersonderzoek blijkt dat de Task Navigation (taak) goed functioneert. Enkele inzichten:

- Gebruikers begrijpen dat er een actie van hen wordt verwacht, dat er een taak openstaat en dat ze hiermee aan de slag moeten.
- Het oranje waarschuwingsicoon met uitroepteken trekt effectief de aandacht naar een naderende deadline.
- Het plaatsen van het oranje waarschuwingsicoon naast de deadline, in plaats van voor de taaknaam, blijkt beter te werken omdat de samenhang duidelijker wordt.

[Onderzoek Generieke Services - Miro](https://miro.com/app/board/uXjVP0OE2fg=/?share_link_id=511269957959)
Miro board dat is gebruikt tijdens het gebruikersonderzoek waarin inzichten te vinden zijn over de Task Navigation.

[GOV.UK Task List - GOV.UK](https://design-system.service.gov.uk/components/task-list/)
Task list implementatie bij GOV.UK waarbij op een vergelijkbare wijze omgegaan wordt met actions/tasks die van de gebruiker verwacht worden. Bij de [Task list Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/97) wordt meer informatie, onderzoek en aanpassingen gedeeld.

[Actionable notifications - Carbon Design System](https://carbondesignsystem.com/components/notification/usage/#actionable-notification)
Actionable notifications in het Carbon Design System, een component dat in grote lijnen vergelijkbaar is met de Task Navigation component.

**Comment by ux-fran** [comment]
## Research nov 2024
Gekeken in diverse files van Amsterdam, Open formulieren, DUO, Open Gemeenten en daar verder iets vergelijkbaars gevonden.

Heb ik wat gemist of weet jij (de lezer) misschien nog een organisatie of andere gemeente met een vergelijkbaar component, vul gerust aan!

**Comment by ux-fran** [comment]
## Gemeente Rotterdam

[Documentatie website](https://nl-design-system.github.io/rotterdam/?path=/story/css-denhaag-action--default)

---

### Screenshots / Varianten

Single action
<img width="740" alt="image" src="https://github.com/user-attachments/assets/b17a3730-8924-41d1-8f95-eddefdfbe2bb" />

Responsive gedrag
<img width="320" alt="image" src="https://github.com/user-attachments/assets/cb903594-2468-436b-a343-d882980d4fa3" />

Mijn Loket
<img width="680" alt="image" src="https://github.com/user-attachments/assets/1313b152-70ac-4359-8ac9-b867b4721845" />


---

### Notities

- Onder de naam: Single action.
- De Action component van Den Haag is hier hergebruikt. Bekijk de [notities van Den Haag](https://github.com/orgs/nl-design-system/discussions/349#discussioncomment-11510661) voor meer details.

**Comment by ux-fran** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/67.2.0/components/layout/action-list)
- [Storybook](https://storybook.dso-toolkit.nl/67.2.0/?path=/story/core-action-list--default)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/action-list)

---


### Screenshots / Varianten

**Storybook**

Default
<img width="800" alt="image" src="https://github.com/user-attachments/assets/5d268c9b-9b01-4da0-9656-ffb52645eb56" />

With warning
<img width="800" alt="image" src="https://github.com/user-attachments/assets/3498bec1-cd27-4a44-89fe-eb355ea83cb0" />

Responsive gedrag
<img width="400" alt="image" src="https://github.com/user-attachments/assets/9a900336-a263-4fb3-98af-97d160c44014" />

---

### Notities

- Onder de naam: Action List.
- Opgebouwd uit `dso-action-list-item`.
- Bestaat uit een lijst met daarin een Step Counter en een Accordion. In deze Accordion staan de taken die uitgevoerd moeten worden. Het is niet zichtbaar of er call-to-actions in staan.
- Er worden geen deadlines weergegeven.
- De content van een Action List Item kan een Heading of een Button bevatten.
- De werking van deze component verschilt van de ‘Action’ component van Den Haag: visueel is hij anders vormgegeven en er is geen directe call-to-action zichtbaar om de taak uit te voeren.

**Reply by eslook** [reply]
Hier zie ik mogelijk meer de [Link List](https://github.com/orgs/nl-design-system/discussions/266) of een [Accordion](https://github.com/orgs/nl-design-system/discussions/217) gebruikt worden.

**Comment by ux-fran** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/css-actions-action--default)
- [Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=10817-61351&t=uz0TWRngCXR1NFzv-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Action)

---

### Screenshots / Varianten

Default
<img width="750" alt="image" src="https://github.com/user-attachments/assets/710db5c4-1972-4fc7-ac0e-62b9e65b87db" />

Hover
<img width="750" alt="image" src="https://github.com/user-attachments/assets/88dd7400-cd5a-4bc9-ac9f-364187504dd2" />

Focus
<img width="750" alt="image" src="https://github.com/user-attachments/assets/31298e3a-d745-424e-93fd-cc407140c607" />

With Date
<img width="750" alt="image" src="https://github.com/user-attachments/assets/15e19599-24b9-4918-b351-9c158785c631" />

With Date Relative
<img width="750" alt="image" src="https://github.com/user-attachments/assets/e80842e1-ef9f-4bea-86f0-01edd10ced52" />

With Date Warning
<img width="750" alt="image" src="https://github.com/user-attachments/assets/3a89f4a2-3b3a-4b44-acbe-e6ccda769113" />

List
<img width="750" alt="image" src="https://github.com/user-attachments/assets/f41105b1-92a8-4492-bb6c-fff32e81c636" />

Responsive gedrag

<img width="400" alt="image" src="https://github.com/user-attachments/assets/fe363b98-1623-4d1b-a960-0fae76383bd1" />

---


### Notities

- Bestaat uit ‘content’ en ‘details’. De ‘content’ bevat de naam van de taak, terwijl de ‘details’ de deadline en de arrow Icon aangeven.
- De gehele component is aanklikbaar.
- Kan al dan niet een deadline bevatten.
- Bij een deadline van minder dan 2 dagen wordt de variant ‘Date Warning’ toegepast.
- Arrow Icon staat altijd aan de rechterkant. Als er geen deadline is, staat deze op small containers in de rechteronderhoek. 
- Richtlijnen beschikbaar.

**Comment by jiromaykin** [comment]
## Open Inwoner 

- [Storybook](https://maykinmedia.github.io/open-inwoner/?path=/docs/components-actionlist--docs )
- [Figma](https://www.figma.com/design/iKGhWhstaLIlFSaND2q7cE/OIP---Designs--new-?node-id=8668-6393&t=pdssBQ1neYgPx8sc-1)
- [GitHub](https://github.com/maykinmedia/open-inwoner/tree/develop/src/open_inwoner/react/components/ActionList)

---

## Afbeelding - extra Status labels:

<img width="807" height="254" alt="OIP-tasknavigation" src="https://github.com/user-attachments/assets/4b76c88c-e14d-46a9-8fdf-dbda18354c35" />

---

## Gebruikersonderzoek 

Onderzoek door Gemeente Groningen 
https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/groningen-mijn-omgeving-homepage/#main
"...**Onderzoeksvraag 2 – Hoe gebruikt een inwoner de homepage?**
De gekleurde labels bij de openstaande acties vallen goed op, en inwoners kennen verschil in urgentie hieraan toe op basis van de kleuren.
Geel/oranje als meest urgent, gevolgd door blauw..."

**Comment by Rozerinay** [comment]
## Verduidelijking van de gekozen naamgeving van 'Action' naar 'Task Navigation'

In [het Help Wanted draadje op Slack is op 14 april 2025](https://codefornl.slack.com/archives/C7KDQRXMZ/p1744627921527609) reageerde de community op de naam: Action. Er kwamen waardevolle punten naar voren, zoals de verwarring met Action Group, de overlap met Card as Link, en de suggestie om ‘Task’ als naam te gebruiken.

Op basis van die input hebben we de naam aangepast naar 'Task Navigation'.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Task Navigation op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=task+navigation)

**Comment by ux-fran** [comment]
## Voorstel: visuele statusindicator in Task navigation (✔ afgerond)

In aanvulling op de bestaande Task navigation wordt een kleine aanpassing voorgesteld: het introduceren van een **visuele statusindicator (vinkje)** vóór de titel van een taak om aan te geven dat deze is afgerond.

### Wat verandert er?

- Afgeronde taken krijgen een **✔-icoon vóór de titel**
- Openstaande taken behouden hun huidige weergave (zonder vinkje)
- De structuur en volgorde van de lijst blijven ongewijzigd

### Waarom deze aanpassing?

Uit het ontwerp en de varianten (zie screenshot) blijkt dat het onderscheid tussen *open* en *afgeronde* taken nu vooral via tabs en tekst wordt gemaakt. In de praktijk:

- Gebruikers scannen lijsten snel en lezen niet altijd labels of tabs  
- De status van een taak is niet direct zichtbaar binnen de lijst zelf  
- Bij langere lijsten kost het extra moeite om status te herkennen  

Door een vinkje toe te voegen:

-  **Wordt de status direct scanbaar**  
-  **Ontstaat er visuele herkenning**  
-  **Sluit het beter aan op mentale modellen** (vinkje = klaar)  

Dit sluit aan bij inzichten uit het MijnPlan-onderzoek. Daarin wordt voorgesteld dat het onderscheid tussen afgeronde en open taken duidelijker zichtbaar moet zijn en dat gebruikers behoefte hebben aan betere feedback op afronding:  
👉 [Rapportage MijnPlan](https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/vng-mijn-plan/#main)

### Hoe ziet dit eruit?

- ✔ Titel taak 1 → afgerond  
- Titel taak 2 → nog open  

(zoals zichtbaar in de bijgevoegde screenshot)

### Gedrag in combinatie met tabs

Deze aanpassing werkt aanvullend op de bestaande tabs (*Open* / *Afgerond*):

- In de tab **Afgerond** bevestigt het vinkje visueel de status  
- In de tab **Open** ontbreken vinkjes → duidelijke consistentie  
- Bij gecombineerde lijsten (bijv. chronologisch) blijft status zichtbaar  

### Overweging

Deze wijziging is klein in implementatie, maar groot in effect op:

- scanbaarheid  
- duidelijkheid  
- consistentie tussen componenten  

---

Benieuwd hoe anderen hiernaar kijken 👀

<img width="278" height="1192" alt="Samenstelling van tabs met open en afgeronde taken." src="https://github.com/user-attachments/assets/c8f83fbf-176a-4a6e-8cd7-e287c3e1ec34" />

**Reply by jiromaykin** [reply]
Dit zou alleen werken voor burgers met slechts enkele zaken. 'Grootgebruikers' zoals ondernemingen krijgen dan in beide Tabs een extreem lange lijst zonder eind. Of dan paginering tonen?
En waarom een extra icoon tonen in de lijst van afgeronde taken? Die krijgen daar allen hetzelfde afgevinkte icoon.

NB: in Open Inwoner staan afgeronde taken op dit moment helemaal niet zichtbaar voor de gebruiker. 
Het zou qua data-traagheid ook veel tijd kosten voor de browser om alle afgeronde en alle openstaande eerst te moeten inladen  - dat moet wel volgens dit design vanwege de getallen die in beeld komen: die kun je alleen 'weten' als browser als alle data is ingeladen.

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨


Help je mee hem door de Candidate stappen te krijgen?
[Task Navigation op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=task+navigation)
